lib/review-apps.unit.test.js

```diff
@@ -217,183 +406,111 @@ describe('review-apps', () => {
-				commit,
-				minTimeout: 0
-			})(appId);
+			expect(true).toBeTruthy();
 
-			expect(waitedReviewAppBuild).toEqual(appId);
 		});
 
-		it('waits for succeeded review app build even with errors', async () => {
+		it('will retry if an error is thrown when attempting to get the app build status', async () => {
+
 			const commit = 'a00000000000';
 			const appId = 'app123';
-			const succeededReviewAppBuild = {
-				app: {
-					id: appId
-				},
-				source_blob: {
-					version: commit
-				},
-				status: 'succeeded'
-			};
-			nockScope
-				.get('/apps/app123/builds')
-				.reply(400, [])
-				.get('/apps/app123/builds')
-				.reply(200, [ succeededReviewAppBuild ]);
 
-			const waitedReviewAppBuild = await waitForReviewAppBuild({
-				commit,
-				minTimeout: 0
-			})(appId);
+			nockScope.get('/apps/app123/builds')
+				.times(60)
+				.reply(400, {
+					message: 'Some error occurred'
+				});
+
+			try {
+				await repeatedCheckForBuildSuccessStatus({ commit, appId, minTimeout: 0 });
+			} catch (error) {
+				expect(error.message).toMatch('Some error occurred');
+				expect(error).toBeInstanceOf(Error);
+			}
 
-			expect(waitedReviewAppBuild).toEqual(appId);
 		});
 
-		it('waits for succeeded review app build from pending', async () => {
+		it('will retry if the build returned does not have a status of \'succeeded\'', async () => {
+
 			const commit = 'a00000000000';
 			const appId = 'app123';
-			const pendingReviewAppBuild = {
+			const reviewAppBuild = {
 				source_blob: {
 					version: commit
 				},
 				status: 'pending'
 			};
-			const succeededReviewAppBuild = {
-				app: {
-					id: appId
-				},
-				source_blob: {
-					version: commit
-				},
-				status: 'succeeded'
-			};
-			nockScope
-				.get('/apps/app123/builds')
-				.reply(200, [ pendingReviewAppBuild ])
-				.get('/apps/app123/builds')
-				.reply(200, [ succeededReviewAppBuild ]);
-
-			const waitedReviewAppBuild = await waitForReviewAppBuild({
-				commit,
-				minTimeout: 0
-			})(appId);
-
-			expect(waitedReviewAppBuild).toEqual(appId);
-		});
 
-		it('fails for failed review app build with output stream link', async () => {
-			const commit = 'a00000000000';
-			const appId = 'app123';
-			const failedReviewAppBuild = {
-				source_blob: {
-					version: commit
-				},
-				status: 'failed',
-				output_stream_url: 'https://heroku.com/builds/output/app123'
-			};
-			nockScope
-				.get('/apps/app123/builds')
-				.reply(200, [ failedReviewAppBuild ]);
+			nockScope.get('/apps/app123/builds')
+				.times(60)
+				.reply(200, [reviewAppBuild]);
 
 			try {
-				await waitForReviewAppBuild({
-					commit,
-					minTimeout: 0
-				})(appId);
-			} catch (e) {
-				const { message } = e;
-				expect(message).toMatch('Review app build failed');
-				expect(message).toMatch('https://heroku.com/builds/output/app123');
+				await repeatedCheckForBuildSuccessStatus({ commit, appId, minTimeout: 0 });
+			} catch (error) {
```

Comment:
Similar to above, is there a way to check how many times the mock has been called? I know jest has https://jestjs.io/docs/en/expect#tohavebeencalledtimesnumber

Reply:
I have used `nock.isDone()`, taken from this blog post: [Ensure All Nock Interceptors Are Used by neverendingqs](https://blog.neverendingqs.com/2017/03/13/ensure-all-nock-interceptors-are-used.html).